lib/core/src/control/mouse_manager.rs

```diff
@@ -0,0 +1,338 @@
+
+/// A struct storing information about mouse move, mouse enter and mouse leave events.
+pub struct MousePositionEvent {
+    pub previous_position : Vector2<f32>,
+    pub position          : Vector2<f32>
+}
+
+impl MousePositionEvent {
+    fn from(event:MouseEvent, data:&Rc<MouseManagerData>) -> Self {
+        let position          = Vector2::new(event.x() as f32,event.y() as f32);
+        let position          = position - data.dom().position();
+        let previous_position = match data.mouse_position() {
+            Some(position) => position,
+            None           => position
+        };
+        data.set_mouse_position(Some(position));
+        Self { previous_position, position }
+    }
+}
+
+
+
+// =======================
+// === MouseWheelEvent ===
+// =======================
+
+pub trait MouseWheelCallback = FnMut(MouseWheelEvent) + 'static;
+
+/// A struct storing information about mouse wheel events.
+pub struct MouseWheelEvent {
+    pub is_ctrl_pressed : bool,
+    pub movement_x      : f32,
+    pub movement_y      : f32
+}
+
+impl MouseWheelEvent {
+    fn from(event:WheelEvent) -> Self {
+        let movement_x      = event.delta_x() as f32;
+        let movement_y      = event.delta_y() as f32;
+        let is_ctrl_pressed = event.ctrl_key();
+        Self { movement_x,movement_y,is_ctrl_pressed }
+    }
+}
+
+
+
+// ==============================
+// === MouseManagerProperties ===
+// ==============================
+
+struct MouseManagerProperties {
+    dom                    : DOMContainer,
+    mouse_position         : Option<Vector2<f32>>,
+    target                 : EventTarget,
+    stop_tracking_listener : Option<MouseEventListener>
+}
+
+
+
+// ========================
+// === MouseManagerData ===
+// ========================
+
+/// A struct used for storing shared MouseManager's mutable data.
+struct MouseManagerData {
+    properties: RefCell<MouseManagerProperties>
+}
+
+impl MouseManagerData {
+    fn new(target:EventTarget, dom:DOMContainer) -> Rc<Self> {
+        let mouse_position         = None;
+        let stop_tracking_listener = None;
+        let p = MouseManagerProperties{dom,mouse_position,target,stop_tracking_listener};
+        let properties = RefCell::new(p);
+        Rc::new(Self { properties })
+    }
+}
+
+
+// === Setters ===
+
+impl MouseManagerData {
+    fn set_mouse_position(&self, position:Option<Vector2<f32>>) {
+        self.properties.borrow_mut().mouse_position = position
+    }
+
+    fn set_stop_mouse_tracking(&self, listener:Option<MouseEventListener>) {
+        self.properties.borrow_mut().stop_tracking_listener = listener;
+    }
+}
+
+
+// === Getters ===
+
+impl MouseManagerData {
+    fn target(&self) -> EventTarget { self.properties.borrow().target.clone() }
+
+    fn mouse_position(&self) -> Option<Vector2<f32>> { self.properties.borrow().mouse_position }
+
+    fn dom(&self) -> DOMContainer { self.properties.borrow().dom.clone() }
```

Comment:
Lets make them multiline, please. They are not aligned nicely here, and its better to just keep it consistent.